### ISSUE TITLE
Directories that used to work now crash in the last few versions

### ISSUE THREAD

**Issue by jt-gilkeson** [UniversalMediaServer/UniversalMediaServer#5189]
The last few versions of UMS have started having this problem, I'll try to open a folder that I used to be able to view without any issue and now I'll get 401 errors.  Fresh install does not fix.

Both Windows x86 and now x86_64 have this problem.  

01:40:24 INFO  Language not specified, defaulting to OS language.
01:40:29 INFO  Starting Universal Media Server 14.8.0
01:40:29 INFO  Based on PS3 Media Server by shagrath, copyright 2008-2014
01:40:29 INFO  https://www.universalmediaserver.com
01:40:29 INFO  
01:40:29 INFO  Build: aeebf4a95 (2024-11-25)

02:08:46 ERROR Exception in result creation "Comparison method violates its general contract!"
java.lang.IllegalArgumentException: Comparison method violates its general contract!
	at java.base/java.util.TimSort.mergeHi(Unknown Source)
	at java.base/java.util.TimSort.mergeAt(Unknown Source)
	at java.base/java.util.TimSort.mergeCollapse(Unknown Source)
	at java.base/java.util.TimSort.sort(Unknown Source)
	at java.base/java.util.Arrays.sort(Unknown Source)
	at java.base/java.util.ArrayList.sort(Unknown Source)
	at java.base/java.util.Collections.sort(Unknown Source)
	at net.pms.store.utils.StoreResourceSorter.sortResourcesByModifiedDate(StoreResourceSorter.java:166)
	at net.pms.store.utils.StoreResourceSorter.sortResourcesByDefault(StoreResourceSorter.java:111)
	at net.pms.store.utils.StoreResourceSorter.sortResourcesByDefault(StoreResourceSorter.java:95)
	at net.pms.store.StoreContainer.sortChildrenIfNeeded(StoreContainer.java:460)
	at net.pms.store.container.VirtualFolder.discoverChildren(VirtualFolder.java:324)
	at net.pms.store.container.VirtualFolder.doRefreshChildren(VirtualFolder.java:373)
	at net.pms.store.StoreContainer.refreshChildren(StoreContainer.java:767)
	at net.pms.store.StoreContainer.refreshChildrenIfNeeded(StoreContainer.java:466)
	at net.pms.store.MediaStore.getResources(MediaStore.java:659)
	at net.pms.network.mediaserver.jupnp.support.contentdirectory.UmsContentDirectoryService.browse(UmsContentDirectoryService.java:729)
	at net.pms.network.mediaserver.jupnp.support.contentdirectory.UmsContentDirectoryService.browse(UmsContentDirectoryService.java:292)
	at jdk.internal.reflect.GeneratedMethodAccessor18.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.jupnp.util.Reflections.invoke(Reflections.java:42)
	at org.jupnp.model.action.MethodActionExecutor.execute(MethodActionExecutor.java:94)
	at org.jupnp.model.action.AbstractActionExecutor$1.execute(AbstractActionExecutor.java:74)
	at org.jupnp.model.DefaultServiceManager.execute(DefaultServiceManager.java:130)
	at org.jupnp.model.action.AbstractActionExecutor.execute(AbstractActionExecutor.java:71)
	at org.jupnp.protocol.sync.ReceivingAction.executeSync(ReceivingAction.java:104)
	at org.jupnp.protocol.ReceivingSync.execute(ReceivingSync.java:65)
	at org.jupnp.protocol.ReceivingAsync.run(ReceivingAsync.java:71)
	at org.jupnp.transport.spi.UpnpStream.process(UpnpStream.java:83)
	at net.pms.network.mediaserver.jupnp.transport.impl.JakartaServletUpnpStream.run(JakartaServletUpnpStream.java:70)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

**Comment by SubJunk** [comment]
@jt-gilkeson thanks for reporting it

@SurfaceS it looks like the `sortResourcesByModifiedDate` method is returning inconsistent results, any ideas?

**Comment by SubJunk** [comment]
@SurfaceS making sure you see this

**Comment by SurfaceS** [comment]
> [@SurfaceS](https://github.com/SurfaceS) making sure you see this

Hard without full log.

I tried with a PR.

**Comment by SubJunk** [comment]
@jt-gilkeson please try this and let us know if it works https://www.universalmediaserver.com/uploads/UMS-5189-x86_64.exe

**Comment by jt-gilkeson** [comment]
Unfortuntaley this is still having the same type of issue on 14.11.0

```
00:16:02 ERROR Exception in result creation "Comparison method violates its general contract!"
java.lang.IllegalArgumentException: Comparison method violates its general contract!
	at java.base/java.util.TimSort.mergeHi(Unknown Source)
	at java.base/java.util.TimSort.mergeAt(Unknown Source)
	at java.base/java.util.TimSort.mergeCollapse(Unknown Source)
	at java.base/java.util.TimSort.sort(Unknown Source)
	at java.base/java.util.Arrays.sort(Unknown Source)
	at java.base/java.util.ArrayList.sort(Unknown Source)
	at java.base/java.util.Collections.sort(Unknown Source)
	at net.pms.store.utils.StoreResourceSorter.sortResourcesByModifiedDate(StoreResourceSorter.java:166)
	at net.pms.store.utils.StoreResourceSorter.sortResourcesByDefault(StoreResourceSorter.java:111)
	at net.pms.store.utils.StoreResourceSorter.sortResourcesByDefault(StoreResourceSorter.java:95)
	at net.pms.store.StoreContainer.sortChildrenIfNeeded(StoreContainer.java:461)
	at net.pms.store.container.VirtualFolder.discoverChildren(VirtualFolder.java:324)
	at net.pms.store.container.VirtualFolder.doRefreshChildren(VirtualFolder.java:373)
	at net.pms.store.StoreContainer.refreshChildren(StoreContainer.java:768)
	at net.pms.store.StoreContainer.refreshChildrenIfNeeded(StoreContainer.java:467)
	at net.pms.store.MediaStore.getResources(MediaStore.java:659)
	at net.pms.network.mediaserver.jupnp.support.contentdirectory.UmsContentDirectoryService.browse(UmsContentDirectoryService.java:760)
	at net.pms.network.mediaserver.jupnp.support.contentdirectory.UmsContentDirectoryService.browse(UmsContentDirectoryService.java:292)
	at jdk.internal.reflect.GeneratedMethodAccessor19.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.jupnp.util.Reflections.invoke(Reflections.java:42)
	at org.jupnp.model.action.MethodActionExecutor.execute(MethodActionExecutor.java:94)
	at org.jupnp.model.action.AbstractActionExecutor$1.execute(AbstractActionExecutor.java:74)
	at org.jupnp.model.DefaultServiceManager.execute(DefaultServiceManager.java:130)
	at org.jupnp.model.action.AbstractActionExecutor.execute(AbstractActionExecutor.java:71)
	at org.jupnp.protocol.sync.ReceivingAction.executeSync(ReceivingAction.java:104)
	at org.jupnp.protocol.ReceivingSync.execute(ReceivingSync.java:65)
	at org.jupnp.protocol.ReceivingAsync.run(ReceivingAsync.java:71)
	at org.jupnp.transport.spi.UpnpStream.process(UpnpStream.java:83)
	at net.pms.network.mediaserver.jupnp.transport.impl.JakartaServletUpnpStream.run(JakartaServletUpnpStream.java:70)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```

**Comment by jt-gilkeson** [comment]
I wonder if the issue is with these lines:
```
	if (asc) {
		return Long.compare(file1.lastModified(), file2.lastModified());
	} else {
		return Long.compare(file2.lastModified(), file1.lastModified());
	}
```

It seems like if the file has the same last modified date that maybe it should go to a second attribute like filename?

**Comment by SurfaceS** [comment]
> It's strange because if I copy a folder that has the issue, the copy opens fine.
> 
> I wonder if the issue has anything to do with these lines:
> 
> ```
> 	if (asc) {
> 		return Long.compare(file1.lastModified(), file2.lastModified());
> 	} else {
> 		return Long.compare(file2.lastModified(), file1.lastModified());
> 	}
> ```
> 
> It seems like if the file has the same last modified date that maybe it should go to a second attribute like filename to get a consistent sort? I dunno, I'm just throwing out guesses here.

I think the only case that can violates its general contract is when both files are null...
PR added #5460

**Comment by jt-gilkeson** [comment]
@SubJunk I can definitely test this new fix if you create a test version.

**Comment by SubJunk** [comment]
@jt-gilkeson cool thanks, here it is https://www.universalmediaserver.com/uploads/UMS-jt-x86_64.exe
Please let me know when you have downloaded it

**Comment by jt-gilkeson** [comment]
@SubJunk  Got it. Fix didn't work, it's till broken.

01:51:04 WARN  Fixing trailing garbage in event message path: /dev/dcb065fa-2665-fab9-b06c-fa0926b92f96/svc/upnp-org/RenderingControl/event/cb192.168.3.209
01:51:04 WARN  Fixing trailing garbage in event message path: /dev/dcb065fa-2665-fab9-b06c-fa0926b92f96/svc/upnp-org/RenderingControl/event/cb192.168.3.209
01:51:04 WARN  Fixing trailing garbage in event message path: /dev/dcb065fa-2665-fab9-b06c-fa0926b92f96/svc/upnp-org/RenderingControl/event/cb192.168.3.209

01:51:04 ERROR Unhandled exception during media scan: Comparison method violates its general contract!

01:51:05 WARN  Fixing trailing garbage in event message path: /dev/dcb065fa-2665-fab9-b06c-fa0926b92f96/svc/upnp-org/RenderingControl/event/cb192.168.3.209
01:51:05 WARN  Fixing trailing garbage in event message path: /dev/dcb065fa-2665-fab9-b06c-fa0926b92f96/svc/upnp-org/RenderingControl/event/cb192.168.3.209
01:51:05 WARN  Fixing trailing garbage in event message path: /dev/dcb065fa-2665-fab9-b06c-fa0926b92f96/svc/upnp-org/RenderingControl/event/cb192.168.3.209
01:51:06 WARN  Fixing trailing garbage in event message path: /dev/dcb065fa-2665-fab9-b06c-fa0926b92f96/svc/upnp-org/RenderingControl/event/cb192.168.3.209
01:51:06 WARN  Fixing trailing garbage in event message path: /dev/dcb065fa-2665-fab9-b06c-fa0926b92f96/svc/upnp-org/RenderingControl/event/cb192.168.3.209
01:51:06 WARN  Fixing trailing garbage in event message path: /dev/dcb065fa-2665-fab9-b06c-fa0926b92f96/svc/upnp-org/RenderingControl/event/cb192.168.3.209
01:51:06 WARN  Fixing trailing garbage in event message path: /dev/dcb065fa-2665-fab9-b06c-fa0926b92f96/svc/upnp-org/RenderingControl/event/cb192.168.3.209
01:51:07 WARN  Fixing trailing garbage in event message path: /dev/dcb065fa-2665-fab9-b06c-fa0926b92f96/svc/upnp-org/RenderingControl/event/cb192.168.3.209
01:51:07 WARN  Fixing trailing garbage in event message path: /dev/dcb065fa-2665-fab9-b06c-fa0926b92f96/svc/upnp-org/RenderingControl/event/cb192.168.3.209

01:51:09 ERROR Exception in PlayerApiServlet: Comparison method violates its general contract!

01:51:34 WARN  Fixing trailing garbage in event message path: /dev/dcb065fa-2665-fab9-b06c-fa0926b92f96/svc/upnp-org/RenderingControl/event/cb192.168.3.209
01:51:34 WARN  Fixing trailing garbage in event message path: /dev/dcb065fa-2665-fab9-b06c-fa0926b92f96/svc/upnp-org/RenderingControl/event/cb192.168.3.209
01:51:34 WARN  Fixing trailing garbage in event message path: /dev/dcb065fa-2665-fab9-b06c-fa0926b92f96/svc/upnp-org/RenderingControl/event/cb192.168.3.209

**Comment by jt-gilkeson** [comment]
TLDR: It looks like the startup scan creates this issue - if I add a new folder (in an existing shared path) while UMS is running, that folder works fine, but when I shut down UMS and restart (and it does the rescan) that's when the folder breaks in UMS.  

Additional Notes (Updated - I discovered this after I tested and added the traces):
There's something weird here, on the last build I made a copy of a folder that wasn't working and that copy folder worked fine. Now that copy folder doesn't work yet I didn't change anything in it.      

If I copy that folder again (while UMS is running) the new folder works fine in UMS.  BUT if I close UMS and restart that copy of a copy folder no longer works!   So maybe this has something to do with the database and startup scan ?!  

Is there any way to just disable the database for UMS so it works on live files like it used to?   That is when UMS started having these issues for me when the database and long startup scan were implemented. 

Either way, hopefully this helps narrow down the search for what broke in modern database-based UMS

**Comment by SurfaceS** [comment]
@jt-gilkeson 

> Is there any way to just disable the database for UMS so it works on live files like it used to? That is when UMS started having these issues for me when the database and long startup scan were implemented.

`sortResourcesByModifiedDate` do not rely on database.
For the startup scan, it discover media as it will do when you browse (it is a pre-browse if you like).
So if a browse fail, the startup scan will fail as well.

The `Comparison method violates its general contract!` is thrown when the result of comparing is, for example :
compare `A` and `B` : `A > B`
compare `B` and `C` : `B > C`
compare `A` and `C` : `A < C` :anguished:

On `sortResourcesByModifiedDate`, it compare `File.lastModified()`.

The only cases I see that can violates the compare is : 
- A file was modified during the compare (compare take ms... so it should not) and the value changed between iteration.
- An I/O error occurs between compare (then returned value is 0).

I presume it is the second one : `I/O error` (it is erratic by essence).

I'll post another PR to take that in account.
It will not fix your I/O errors, but UMS will warn you about it, and give back the medias unsorted instead of throw without results.

**Comment by SurfaceS** [comment]
@jt-gilkeson 
Added a PR #5468 with test
Please let me know when you have downloaded it

**Comment by jt-gilkeson** [comment]
Installed your build.   
The fallback behavior you implemented works, but like I said the failures only occur on these folders after a restart of the app - I'm not sure how I/O errors would come out of that process.    It happens in a number of folders.    

None of the new traces you added occur (No "getLastModifiedTime thrown" or "Invalid path for")  

Of note my sorting settings have:

Ignore articles "a" and "the" while sorting

Hide file extensions

both checked.  

I also have show the #transcode folder enabled. 


only the traces happen.  
TRACE 2025-04-17 06:18:02.939 [Media Scanner] net.pms.store.utils.StoreResourceSorter sortResourcesByModifiedDate error: Comparison method violates its general contract!

I'm seeing some weird MediaInfoStore errors - but they don't appear to be in the folders that have issues:

TRACE 2025-04-17 06:16:38.845 [Media Scanner] net.pms.store.MediaInfoStore Store does not yet contain MediaInfo for I:\ToBeMoved\friends - real life\anothername\sub\File Name.m4a
ERROR 2025-04-17 06:16:39.016 [Media Scanner] net.pms.store.MediaInfoStore Error in RealFile.resolve: Unable to find next atom because identifier is invalid     
TRACE 2025-04-17 06:16:39.016 [Media Scanner] net.pms.store.MediaInfoStore 
org.jaudiotagger.audio.exceptions.NullBoxIdException: Unable to find next atom because identifier is invalid     
	at org.jaudiotagger.audio.mp4.atom.Mp4BoxHeader.update(Mp4BoxHeader.java:154)
	at org.jaudiotagger.audio.mp4.atom.Mp4BoxHeader.seekWithinLevel(Mp4BoxHeader.java:356)
	at org.jaudiotagger.audio.mp4.Mp4TagReader.read(Mp4TagReader.java:118)
	at org.jaudiotagger.audio.mp4.Mp4FileReader.getTag(Mp4FileReader.java:48)
	at org.jaudiotagger.audio.generic.AudioFileReader2.read(AudioFileReader2.java:61)
	at org.jaudiotagger.audio.AudioFileIO.readFile(AudioFileIO.java:356)
	at org.jaudiotagger.audio.AudioFileIO.read(AudioFileIO.java:192)
	at net.pms.parsers.JaudiotaggerParser.parse(JaudiotaggerParser.java:198)
	at net.pms.parsers.MediaInfoParser.parseFileForAudioMetadata(MediaInfoParser.java:1271)
	at net.pms.parsers.MediaInfoParser.parse(MediaInfoParser.java:563)
	at net.pms.parsers.Parser.parse(Parser.java:81)
	at net.pms.store.MediaInfoStore.getMediaInfo(MediaInfoStore.java:152)
	at net.pms.store.item.RealFile.resolve(RealFile.java:254)
	at net.pms.store.StoreContainer.addChild(StoreContainer.java:122)
	at net.pms.store.container.VirtualFolder.manageFile(VirtualFolder.java:112)
	at net.pms.store.container.VirtualFolder.analyzeChildren(VirtualFolder.java:200)
	at net.pms.store.container.VirtualFolder.discoverChildren(VirtualFolder.java:325)
	at net.pms.store.MediaScanner.scan(MediaScanner.java:141)
	at net.pms.store.MediaScanner.scan(MediaScanner.java:147)
	at net.pms.store.MediaScanner.scan(MediaScanner.java:147)
	at net.pms.store.MediaScanner.scan(MediaScanner.java:147)
	at net.pms.store.MediaScanner.startScan(MediaScanner.java:98)
	at net.pms.store.MediaScanner.lambda$startMediaScan$0(MediaScanner.java:198)


ERROR Error in RealFile.resolve: Cannot invoke "java.lang.Long.intValue()" because the return value of "net.pms.parsers.mediainfo.StreamVideo.getWidth(net.pms.parsers.mediainfo.MediaInfoHelper, int)" is null
06:16

**Comment by jt-gilkeson** [comment]
More info:  I turned off 
Ignore articles "a" and "the" while sorting
Hide file extensions
show the #5315 

and now everything is working without errors.  I'm going to try turning them on one by one and see which one is the problem (I'm thinking maybe the transcode)

**Comment by jt-gilkeson** [comment]
Just confirmed 

It's the  `show the #transcode folder enabled` setting - that blows up the sort for some reason.   With that off sort works perfectly with no errors, with it on errors galore.

**Comment by SurfaceS** [comment]
@jt-gilkeson 
do not sort transcode folder (already sorted by engine)
catch NullBoxIdException on JaudiotaggerParser

you can test new one

**Comment by jt-gilkeson** [comment]
Tried the latest, it did Not fix the sort when this setting is on.

still generating:  
net.pms.store.utils.StoreResourceSorter sortResourcesByModifiedDate error: Comparison method violates its general contract!
and falling back to unsorted when  "Show the "#--TRANSCODE--#" folder" is on.  

If I turn that setting off and restart UMS, sorting works.

**Comment by jt-gilkeson** [comment]
Ok this is interesting, in order to preserve some privacy and make the logs smaller I changed my config to just use a test folder with the known bad folder in it ("name": "spaz - Copy - Copy")

Here's what happened:

"Show the "#--TRANSCODE--#" folder" on
Opened my known bad folder, sort failed order is wrong (but NO contract violation was reported in the log - so maybe that is a separate but related issue?)

"Show the "#--TRANSCODE--#" folder" off (did not restart - just changed the option) 
Opened my known bad folder, sort succeeded  

Attached is the log trace, you'll see the order of files has changed between the first time displaying the folder and second time displaying the folder -  yet no sorting changes were made, only the show transcode setting was toggled off.    

[good--ums_dbg_2025-04-18-01-16.zip](https://github.com/user-attachments/files/19810271/good--ums_dbg_2025-04-18-01-16.zip)

**Comment by jt-gilkeson** [comment]
OK switched the folder to a different one that does do the contract violation.   ("name": "Jeanna")

"Show the "#--TRANSCODE--#" folder" off
Opened my known bad folder, sort succeeded  (j-good)

"Show the "#--TRANSCODE--#" folder" on - didnt't mess up sort until I restarted UMS
TRACE 2025-04-18 01:46:19.603 [webplayer-server-72] net.pms.store.utils.StoreResourceSorter sortResourcesByModifiedDate error: Comparison method violates its general contract!
Bad sort. (j-bad)

[j-good-ums_dbg_2025-04-18-01-43.zip](https://github.com/user-attachments/files/19810490/j-good-ums_dbg_2025-04-18-01-43.zip)

[j-bad-ums_dbg_2025-04-18-01-46.zip](https://github.com/user-attachments/files/19810492/j-bad-ums_dbg_2025-04-18-01-46.zip)

**Comment by SurfaceS** [comment]
@jt-gilkeson 

Strange thing is that it throw on the root folder, not on the `#--TRANSCODE--#` content.
The last one is handling non SystemFileResource.
You can test it [here](https://1fichier.com/?ai4jp1neq6vh4l38g46x).

**Comment by jt-gilkeson** [comment]
Success!  
Nice work. I can toggle the transcode on and off now and sorting does not break - I get the correct sort every time.
No contract violations in the log.  

I've been using this project since the PMS days on my Fat PS3 which is still my main Blu-ray and home media device.  Happy to finally have helped in the process of improving it.  

As an addendum now that the main issue is fixed,  one thing I've noticed (in this build anyways) which I'm not sure whether it's expected behavior or not is that the Transcode folder itself if not sorted in the sort order - so if I'm sorting by date and my first entry in the normal folder is XYZ.mpg, if I go into the transcode folder that is not the first item, it's in a different order (maybe unsorted) .  It seems like the order should match, so it's easy to find the video that you want to transcode - I *Think* it used to work that way - i.e. the order was the same in both folders.

**Comment by SurfaceS** [comment]
@jt-gilkeson I [implemented](https://1fichier.com/?0mdy3f297194d9zfe3hr) it, but I don't think it was sorting the #--TRANSCODE--# children as it was :
- set to not sort children (`setSortable(false)` is for itself, `setChildrenSorted(true/false, false as default)` for children)
- FileTranscodeVirtualFolder was not exposing the `File`.

**Comment by SubJunk** [comment]
@jt-gilkeson cool that you have been using UMS for so long! Also with those fat PS3s usually the soldering comes loose and needs to be re-melted, did you do the heat gun trick or did you get lucky? 😆 I had to re-solder mine twice before I got a PS3 Slim

**Comment by jt-gilkeson** [comment]
@SubJunk I have a CECHK01 (65nm CPU and 65nm RSX - so not the oldest of the Fat) - I've never opened it (beyond pulling out the hard drive chassis to upgrade) so I've been lucky so far.  I do need to change the paste at some point.  

@SurfaceS That did the trick, the transcode folder order now matches the main folder order.

**Comment by Mik-S-UMS** [comment]
@SubJunk I had to do the heatgun trick several times on my day one PS3 but never lasted more then a few weeks at most so I got a slim and did the trick one more time to be able to transfer all the data to the new one.

The heatgun trick does not actually melt the solder to do a reflow, it just softens the epoxy around the silicon die so it allows it to move back into position and make better contact. 

The cause of this is the epoxy holding the silicone die on the substrate of the RSX has unsuitable thermal expansion for this purpose and causes the solder bumps on the bottom of the die to become detached, these are not the solder balls on the bottom of the RSX that connect to the PCB. This is also the same cause for the Xbox RROD and some Nvidia GPUs failing from around that time. 

It could also be the NEC-TOKIN caps that are failing and these were blamed for a while but was easily mistaken due to the heat required to replace them that would fix the RSX issue, but it usually does not last. It was not until the UART connection to the syscon was discovered that it would say what the failure actually was.

[RIP Felix](https://www.youtube.com/watch?v=I0UMG3iVYZI) did a lot figuring this out and has made some great videos going over all these discoveries and how to properly fix this. Basically the only proper fix is a Frankie, replacing the 65mn RSX in the Fat with a 40nm from a Slim but only really worth it on one of the back-compatible versions. Unfortunately my fat was not HW back-compat as they removed the PS2 chip before the UK launch so I never got it fixed.

**Comment by jt-gilkeson** [comment]
One last semi-related possible bug I'm seeing is this:
On my PS3 if I open UMS for images, only images are shown in a folder  (correct behavior), but there is also a transcode folder showing (if video files exit in the source folder - even though they are screened out in the "image mode" browsing), if I open the transcode folder  shows me entries for video files (not listed in the main folder - because we are in "image mode") and if I open the individual video transcode folder there are no entries in it.

It seems like:
a) transcode shouldn't be shown when you are browsing images
or (if there is some reason to show)
b) entries for video files shouldn't be shown in "image mode", and entries that have 0 actual transcode options underneath them shouldn't be shown.  

Maybe this should be a separate issue, but it seems like it's in the same area of these other issues.

**Comment by SubJunk** [comment]
Cool details about the soldering! Man that machine is great for Blu-rays. Such good bang for buck.

I don't think we have any control over that last thing you mentioned @jt-gilkeson because of the way the PS3 does those folders. Basically it is receiving our full list of folders all 3 times (images, photos and videos) but it is just hiding the things that don't fit those categories. UMS doesn't know the PS3 is doing that

**Comment by jt-gilkeson** [comment]
@SubJunk  Got it, so those "video file" entries in the transcode folder are just folders (as far as PS3 is concerned) and the contents of each folder (the various transcode variants) are being filtered out by the PS3 because they are not image files but video streams/files. That makes sense.  Thanks for the info.
